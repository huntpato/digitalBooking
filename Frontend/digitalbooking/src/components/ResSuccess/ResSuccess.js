import React from 'react'
import { Link } from "react-router-dom";
import checked from '../../assets/checked.png'
import styles from  './ResSuccess.module.css'

export const ResSuccess = (props) => {
    const {confirmation, success, title, info, button} = styles;
    const { message, text, bMessage } = props;
    return (
        <div className= { confirmation }>
            <section className= { success }>
                <img src= { checked } alt='Ícono de reserva realizada.'/>
                <h3 className= { title }>{ message }</h3>
                <h5 className= { info }>{text}</h5>
                <Link to= '/'><button className={ button }>{ bMessage }</button></Link>
            </section>
        </div>
    )
}
