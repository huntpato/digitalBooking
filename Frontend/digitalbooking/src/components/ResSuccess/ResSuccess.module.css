.confirmation {
  align-items: center;
  background-color: var(--background-color2);
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 93vh;
  width: 100%;
}

.success {
  align-items: center;
  background-color: var(--background-white);
  border: 1px solid #dfe4ea;
  box-shadow: 0px 3px 4px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 45%;
  margin: 40px 0 0 0;
  width: 40%;
}

.confirmation img {
  height: 55px;
  position: relative;
  top: 0px;
}

.title {
  color: var(--principal-color);
  font-size: 20px;
  margin-top: 15px;
}

.info {
  color: var(--color-typo);
  font-size: 16px;
  padding: 6px 0 20px 0;
}

.button {
  background-color: var(--principal-color);
  border-radius: 5px;
  border: none;
  color: var(--background-white);
  cursor: pointer;
  font-weight: 600;
  height: 30px;
  padding: 0 80px 0 80px;
}

.button:hover {
  background-color: var(--principal-color);
  color: var(--background-white);
  opacity: 2;
}

.button:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.98);
}


@media (max-width: 1200px){

  .success {
    width: 45%;
  }

}

@media (max-width: 768px){

  .success {
    height: 38%;
    width: 75%;
  }

}

@media (max-width: 480px){

  .success {
    width: 85%;
  }

}