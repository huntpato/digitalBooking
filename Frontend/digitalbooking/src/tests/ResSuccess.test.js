import React from 'react';
import { shallow } from 'enzyme';
import { ResSuccess } from '../components/ResSuccess/ResSuccess'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <ResSuccess />', ()=> {

    const wrapper= shallow(<ResSuccess />);

    test('Comprobar que <ResSuccess /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })


    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('confirmation')).toBe(true);
    })

})
