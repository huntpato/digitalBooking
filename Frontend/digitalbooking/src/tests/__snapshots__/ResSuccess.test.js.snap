// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ResSuccess /> Comprobar que <ResSuccess /> renderiza correctamente. 1`] = `
<div
  className="confirmation"
>
  <section
    className="success"
  >
    <img
      alt="Ícono de reserva realizada."
      src="checked.png"
    />
    <h3
      className="title"
    />
    <h5
      className="info"
    />
    <Link
      to="/"
    >
      <button
        className="button"
      />
    </Link>
  </section>
</div>
`;
